src/hotspot/share/ci/ciStreams.hpp

```diff
@@ -249,6 +249,16 @@ class ciBytecodeStream : StackObj {
            tag.is_dynamic_constant_in_error();
   }
 
+  bool is_string_constant() const {
+    assert(cur_bc() == Bytecodes::_ldc    ||
+           cur_bc() == Bytecodes::_ldc_w  ||
+           cur_bc() == Bytecodes::_ldc2_w, "not supported: %s", Bytecodes::name(cur_bc()));
+
+    int index = get_constant_pool_index();
+    constantTag tag = get_raw_pool_tag(index);
```

Comment:
Looks like these lines are the same as in `is_dynamic_constant()`. Can you move them in separate method to avoid duplication?

Reply:
Ok, I introduced a no-arg `get_raw_pool_tag()` helper method. Does it look better?